src/serialization/ObjectData.ts

```diff
@@ -561,8 +561,8 @@ export class ObjectDataInput implements DataInput {
     }
 
     private assertAvailable(numOfBytes: number, pos: number = this.pos): void {
-        assert(pos >= 0);
-        assert(pos + numOfBytes <= this.buffer.length);
+        assertCond(pos >= 0);
+        assertCond(pos + numOfBytes <= this.buffer.length);
```

Comment:
Looking at the Java side, these seem to be regular checks, not asserts. So, they are checked in the release builds as well